realm_export: Add support to create "full data export" via `/export/realm`.

prakhar1144 opened the pull request:
API changes to make it possible to use `POST /export/realm` for "full data export with member consent". Earlier, only public data export was supported.

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

prakhar1144 commented on zerver/views/realm_export.py:
```diff
@@ -41,19 +57,20 @@ def export_realm(request: HttpRequest, user: UserProfile) -> HttpResponse:
     )
+    if export_type == RealmExport.EXPORT_PUBLIC:
+        realm_count_query.filter(subgroup="public_stream")
+    exportable_messages_estimate = sum(realm_count.value for realm_count in realm_count_query)
```
In the case of "full export with user consent", this will consider all the messages sent, irrespective of the users who consented or not. Is it fine -- do maxm users consent in such export cases?

timabbott replied:
Yeah, that's fine -- this is just an estimate.

timabbott commented on zerver/openapi/zulip.yaml:
```diff
@@ -21737,6 +21776,16 @@ components:
+            - 1 = Public data export.
+            - 2 = Full data export with member consent.
+
+            **Changes**: New in Zulip 10.0 (feature level 300).
```
This changes entry should continue with "Previously, only public data exports were supported via this API endpoint."

timabbott replied:
Oh, actually, this is a case where it should be "Previously, the export type was not included in these objects. While only public data exports could be created via the API or UI, it was possible for other exports done by server administrators to be listed here."

prakhar1144 replied:
Added:
> Previously, the export type was not included in these objects because only public data exports could be created or listed via the API or UI.

This is not correct, right ?
> it was possible for other exports done by server administrators to be listed here.

timabbott replied:
I think we had a small number that appeared there across all of Zulip Cloud, but it is probably irrelevant as a detail, so maybe not worth mentioning.

timabbott commented:
Generally looks fine; posted some notes on documentation and we should decide if we want to use strings in the API in the "API design" conversation.

prakhar1144 commented:
Addressed the review comments.

We still need to make a final decision on whether to use string or int. I can propose a few strings if we decide to do so...

timabbott commented:
Merged, thanks @prakhar1144!
